roles/manage_transit_peered_networks/tasks/create.yml

```diff
@@ -403,11 +419,27 @@
           - private
       when: existing_priv_network_instance.instances is defined and existing_priv_network_instance.instances | length > 0
 
+    - name: Lookup private network instance ami 
+      amazon.aws.ec2_ami_info:
+        region: "{{ aws_region }}"
+        owners: "{{ priv_network_instance_ami_owner }}"
+        filters:
+          name: "{{ priv_network_instance_ami_filter }}"
+          architecture: "{{ priv_network_instance_ami_architecture }}"
+      register: private_network_amis
+      when: priv_network_instance_ami is not defined
+
+    - name: Select private network instance ami
+      set_fact:
+        priv_network_instance_ami: >
+          {{ (private_network_amis.images | selectattr('name', 'defined') | sort(attribute='creation_date'))[-1].image_id }}
+      when: priv_network_instance_ami is not defined
+
     - name: Provision a test VM in the private network
       amazon.aws.ec2_instance:
         count: 1
         image:
-          id: "{{ priv_network_instance_ami }}"
+          id: "{{ priv_network_instance_ami | trim }}"
```

Comment:
What do you think about moving the trim statements up to the assignment line on 435?